torch/csrc/distributed/c10d/FlightRecorder.cpp

```diff
@@ -289,25 +278,20 @@ std::vector<FlightRecorder::Entry> FlightRecorder::dump_entries() {
       result.end(),
       entries_.begin(),
       entries_.begin() + static_cast<std::ptrdiff_t>(next_));
-  // query any remaining events
-  for (auto& r : result) {
-    update_state(r);
-    r.start_ = r.end_ = nullptr;
-  }
   return result;
 }
 
-// Returns the entry with the given id, if it exists. Otherwise, returns
-// std::nullopt.
-std::optional<FlightRecorder::Entry> FlightRecorder::getEntry(
+// Returns the entry's pointer with the given id, if it exists. Otherwise,
+// returns std::nullopt.
+std::optional<FlightRecorder::Entry*> FlightRecorder::getEntry(
     std::optional<size_t> id) {
   if (!enabled_ || !id) {
     return std::nullopt;
   }
 
   std::unique_lock<std::mutex> guard(mutex_);
```

Comment:
Yeah, we should unify the code. In most cases they are the same. `unique_lock` will release the mutex when it goes out of scope.

Reply:
I think will deadlock if you just move it up - since `getEntry` also attempts to grab the same `mutex_`. Maybe need some refactoring to pass the lock? Or remove lock from `getEntry` and assume that it can only be called with a lock held?